Framework/SINQ/src/PoldiPeakSearch.cpp

```diff
@@ -113,16 +113,9 @@ std::list<MantidVec::const_iterator> PoldiPeakSearch::findPeaks(MantidVec::const
    */
   rawPeaks.sort(&PoldiPeakSearch::vectorElementGreaterThan);
 
-  size_t usedPeakCount = std::min(m_maximumPeakNumber, static_cast<int>(rawPeaks.size()));
-  std::list<MantidVec::const_iterator> truncatedPeaks;
+  auto usedPeakCount = std::next(rawPeaks.begin(), std::min(m_maximumPeakNumber, static_cast<int>(rawPeaks.size())));
```

Comment:
```suggestion
  size_t usedPeakCount =  std::min(m_maximumPeakNumber, static_cast<int>(rawPeaks.size()));
  auto lastPeakIter =  std::next(rawPeaks.begin(), usedPeakCount);
```
Just to make this a bit clearer